src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1589,11 +1589,16 @@ def _install_dev_spaces_extension(extension_name):
     return True
 
 
-def _update_dev_spaces_extension(extension_name):
+def _update_dev_spaces_extension(extension_name, extension_module):
     from azure.cli.core.extension import ExtensionNotInstalledException
     try:
+        # unloading the imported module to update
+        del sys.modules[extension_module]
```

Comment:
This is not good as it does not update any references to the old module. Also, a re-import will force python to to an expensive full import.

The way to do this more safely (though still not usually advised) is to use [reload()](https://docs.python.org/3/library/importlib.html#importlib), which will update the module currently in `sys.modules` and allow existing references to have access to the updated module. It is also more performant.

Let's discuss this offline and see how we can address your scenario in the extension.